Userland/Applications/PixelPaint/Layer.cpp

```diff
@@ -299,17 +299,33 @@ void Layer::update_cached_bitmap()
     }
 }
 
-ErrorOr<void> Layer::create_mask()
+ErrorOr<void> Layer::create_mask(MaskType type)
 {
-    m_mask_bitmap = TRY(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRx8888, size()));
-    m_mask_bitmap->fill(Gfx::Color::White);
+    m_mask_type = MaskType::None;
+
+    switch (type) {
+    case MaskType::BasicMask:
+        m_mask_bitmap = TRY(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRx8888, size()));
+        m_mask_bitmap->fill(Gfx::Color::White);
+        break;
+    case MaskType::AlphaMask:
+        m_mask_bitmap = TRY(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, size()));
```

Comment:
Does this need to be a full RGBA image? Seems like a single 8-bit channel should be enough, if that's possible.

Reply:
For the current implementation it is necsessary to have RGBA as my intention for the "Alpha Mask" was really the focus on the alpha channel. Where the "Basic Mask" currently is using the colors that are converted to grey-scale and derived a alpha value from that. So i thougt its better to keep that behaviour clearly separated.

Additionaly it was the easy path to sketch the feature as i could just use all our painting tools as is without taking care of the color that the user has selected.

Reply:
Understood & agreed, it's indeed way simpler to do it that way for now :^)